### PR TITLE
create a guide-variable for whiptail-menu's

### LINKED ISSUE #1477: whiptail-menu: instruct how to use it
as the title says

### PR DESCRIPTION
**Pull request by szaimen**
fix #1477
problem was we had MENU_GUIDE already. I renamed it to RUN_LATER_GUIDE and used the old name MENU_GUIDE for the new whiptail-menu guide

### CODE REVIEW

**Comment by szaimen** [comment]
Todo: depending on if you merge this PR or the smb-server PR first, I need to add this new variable to the SMB-server script here or there.

**Comment by enoch85** [comment]
Why not just include this text in the MENU_GUIDE?

**Comment by szaimen** [comment]
> Why not just include this text in the MENU_GUIDE?

Because I wan't to be able to separate both texts: I would suggest using the RUN_LATER_GUIDE only in scripts in the menu-folder and the MENU_GUIDE for all whiptail --menu's.

File: lib.sh [hunk 1]
```diff
@@ -77,7 +77,8 @@ use_global_systemd_dns
 TITLE="Nextcloud VM - $(date +%Y)"
 [ -n "$SCRIPT_NAME" ] && TITLE+=" - $SCRIPT_NAME"
 CHECKLIST_GUIDE="Navigate with the [ARROW] keys and (de)select with the [SPACE] key. Confirm by pressing [ENTER]"
-MENU_GUIDE="You can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"
+MENU_GUIDE="Navigate with the [ARROW] keys and confirm by pressing [ENTER]"
+RUN_LATER_GUIDE="You can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"
```
Comment (szaimen) [inline lib.sh:81]: BTW: I am referring to those variables

**Comment by enoch85** [comment]
...but you can reach all menus with menu.sh?

**Comment by szaimen** [comment]
> ...but you can reach all menus with menu.sh?

I am not sure if I get what you mean? 

Yes, of course you can also reach e.g. bitwarden_registration.sh and bitwarden-rs_admin-panel.sh with menu.sh but I think that the run_later_guide variable fits only for all Menus (no matter if whiptail-checklist or whiptail-menu) inside the menu folder and nowhere else ...

**Comment by enoch85** [comment]
> Yes, of course you can also reach e.g. bitwarden_registration.sh and bitwarden-rs_admin-panel.sh with menu.sh but I think that the run_later_guide variable fits only for all Menus (no matter if whiptail-checklist or whiptail-menu) inside the menu folder and nowhere else ...

I don't agree, I think the text in `run_later_guide` should be included in `menu_guide` since you can reach all scripts from the menu. 

One less variable, and cleaner IMHO. Also, more informative since it's not selective.

**Comment by szaimen** [comment]
But shouldn't it in this case also get included in all whiptail-checklist's via CHECKLIST_GUIDE?
I guess in this case it would make sense to do something like:
```
CHECKLIST_GUIDE="Navigate with the [ARROW] keys and (de)select with the [SPACE] key. Confirm by pressing [ENTER]\nYou can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"
MENU_GUIDE="Navigate with the [ARROW] keys and confirm by pressing [ENTER]\nYou can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"
```

**Comment by szaimen** [comment]
The problem that I see with including this text in just one variable is, that I cannot separate the guide from the run_later_guide. So e.g. in the smbserver script I had to make my own guide since the run_later_guide wouldn't apply there.

**Comment by szaimen** [comment]
What do we do here?

**Comment by enoch85** [comment]
I think we should have one default variable that's included in all whiptails, which instructs the user how to navigate and select.

If the procedure is different, then add a variable for that. I like the current setup, but don't want to remove something, rather add instructions.

We should form a standard depending on which menu that is used. Something like this:

- Checklist = `$CHECKLIST_GUIDE\n\n$RUN_LATER_GUIDE`
- Menu = `MENU_GUIDE\n\n$RUN_LATER_GUIDE`
- Radio button = `$CHECKLIST_GUIDE\n\n$RUN_LATER_GUIDE`

So we shouldn't replace ``MENU_GUIDE``, but reformat the text so that ``RUN_LATER_GUIDE`` can be used everywhere.

**Comment by szaimen** [comment]
I think this is what I actually implemented here, no? Have you looked at the codechanges?
BTW: we don't have a whiptail-radiolist anywhere now (since we substituted it by using whiptail-menu's) so we don't need a variable for that.

File: apps/bitwarden-rs_admin-panel.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ then
 fi
 
 # Yes or No?
-choice=$(whiptail --title "$TITLE" --menu "Do you want to disable the Bitwarden_rs admin-panel?\nYou can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "Do you want to disable the Bitwarden_rs admin-panel?\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/bitwarden-rs_admin-panel.sh:31]: You don't use both here as suggested.

File: apps/additional_apps.sh [hunk 3]
```diff
@@ -18,7 +18,7 @@ debug_mode
 root_check
 
 # Install Apps
-choice=$(whiptail --title "$TITLE" --checklist "Which apps do you want to install?\n\nAutomatically configure and install selected apps\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Which apps do you want to install?\n\nAutomatically configure and install selected apps\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/additional_apps.sh:21]: You don't use both here as suggested.

File: apps/bitwarden_registration.sh [hunk 4]
```diff
@@ -38,7 +38,7 @@ else
 fi
 
 # Yes or No?
-choice=$(whiptail --title "$TITLE" --menu "Do you want to disable Bitwarden User Registration?\nYou can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "Do you want to disable Bitwarden User Registration?\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/bitwarden_registration.sh:41]: You don't use both here as suggested.

File: apps/collabora_docker.sh [hunk 5]
```diff
@@ -31,7 +31,7 @@ check_nextcloud_https "Collabora (Docker)"
 print_text_in_color "$ICyan" "Checking if Collabora is already installed..."
 if does_this_docker_exist 'collabora/code'
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Collabora' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Collabora' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/collabora_docker.sh:34]: You don't use both here as suggested.

File: apps/collabora_integrated.sh [hunk 6]
```diff
@@ -81,7 +81,7 @@ fi
 # Check if Collabora is installed using the new method
 if is_app_enabled richdocumentscode
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Collabora' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Collabora' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/collabora_integrated.sh:84]: You don't use both here as suggested.

File: apps/fail2ban.sh [hunk 7]
```diff
@@ -28,7 +28,7 @@ lowest_compatible_nc 13
 print_text_in_color "$ICyan" "Checking if Fail2Ban is already installed..."
 if is_this_installed fail2ban && [ -f "/etc/fail2ban/filter.d/nextcloud.conf" ]
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Fail2Ban' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Fail2Ban' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/fail2ban.sh:31]: You don't use both here as suggested.

File: apps/fulltextsearch.sh [hunk 8]
```diff
@@ -34,7 +34,7 @@ cpu_check 2 FullTextSearch
 print_text_in_color "$ICyan" "Checking if Fulltextsearch is already installed..."
 if does_this_docker_exist "$nc_fts"
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Fulltextsearch' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Fulltextsearch' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/fulltextsearch.sh:37]: You don't use both here as suggested.

File: apps/netdata.sh [hunk 9]
```diff
@@ -21,7 +21,7 @@ root_check
 print_text_in_color "$ICyan" "Checking if Netdata is already installed..."
 if [ -d /etc/netdata ]
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Netdata' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Netdata' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/netdata.sh:24]: You don't use both here as suggested.

File: apps/onlyoffice_docker.sh [hunk 10]
```diff
@@ -38,7 +38,7 @@ fi
 print_text_in_color "$ICyan" "Checking if Onlyoffice Docker is already installed..."
 if does_this_docker_exist 'onlyoffice/documentserver'
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Onlyoffice Docker' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Onlyoffice Docker' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/onlyoffice_docker.sh:41]: You don't use both here as suggested.

File: apps/onlyoffice_integrated.sh [hunk 11]
```diff
@@ -89,7 +89,7 @@ elif version_gt "$CURRENTVERSION" "18.0.1" && ! does_this_docker_exist 'onlyoffi
 then
     if is_app_enabled documentserver_community
     then
-        choice=$(whiptail --title "$TITLE" --menu "It seems like 'OnlyOffice' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+        choice=$(whiptail --title "$TITLE" --menu "It seems like 'OnlyOffice' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/onlyoffice_integrated.sh:92]: You don't use both here as suggested.

File: apps/smbmount.sh [hunk 12]
```diff
@@ -752,7 +752,7 @@ return
 while :
 do
     # Main menu
-    choice=$(whiptail --title "$TITLE" --menu "This script let you manage SMB-shares to access files from the host-computer or other machines in the local network.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "This script let you manage SMB-shares to access files from the host-computer or other machines in the local network.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
     "Add a SMB-mount" "(and mount/connect it)" \
     "Mount SMB-shares" "(connect SMB-shares)" \
     "Show all SMB-mounts" "(show detailed information about the SMB-mounts)" \
```
Comment (enoch85) [inline apps/smbmount.sh:758]: You don't use both here as suggested.

File: apps/talk.sh [hunk 13]
```diff
@@ -38,7 +38,7 @@ lowest_compatible_nc 13
 print_text_in_color "$ICyan" "Checking if Talk is already installed..."
 if [ -n "$(occ_command_no_check config:app:get spreed turn_servers | sed 's/\[\]//')" ] || is_this_installed coturn
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Nextcloud Talk' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Nextcloud Talk' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline apps/talk.sh:41]: You don't use both here as suggested.

File: apps/talk_signaling.sh [hunk 14]
```diff
@@ -43,7 +43,7 @@ check_nextcloud_https "Nextclod Talk"
 print_text_in_color "$ICyan" "Checking if Talk is already installed..."
 if [ -n "$(occ_command_no_check config:app:get spreed turn_servers | sed 's/\[\]//')" ] || is_this_installed coturn
 then
-    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Nextcloud Talk' is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like 'Nextcloud Talk' is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
     "Reinstall Nextcloud Talk" "" \
```
Comment (enoch85) [inline apps/talk_signaling.sh:47]: You don't use both here as suggested.

File: lib.sh [hunk 15]
```diff
@@ -311,7 +312,7 @@ explainer_popup() {
 }
 
 reinstall_remove_menu() {
-    choice=$(whiptail --title "$TITLE" --menu "It seems like $SCRIPT_NAME is already installed.\nChoose what you want to do." "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE" --menu "It seems like $SCRIPT_NAME is already installed.\nChoose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline lib.sh:315]: You don't use both here as suggested.

File: menu/additional_apps.sh [hunk 16]
```diff
@@ -26,7 +26,7 @@ else
 fi
 
 # Install Apps
-choice=$(whiptail --title "$TITLE" --checklist "Which apps do you want to install?\n\nAutomatically configure and install selected apps\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Which apps do you want to install?\n\nAutomatically configure and install selected apps\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/additional_apps.sh:29]: You replace menu_guide with run_later_guide here.

You don't use both here as suggested.

File: menu/bitwarden_menu.sh [hunk 17]
```diff
@@ -25,7 +25,7 @@ else
     STARTUP_SWITCH="OFF"
 fi
 
-choice=$(whiptail --title "$TITLE" --checklist "Automatically configure and install the Bitwarden or configure some aspects of it.\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Automatically configure and install the Bitwarden or configure some aspects of it.\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/bitwarden_menu.sh:28]: You don't use both here as suggested.

File: menu/documentserver.sh [hunk 18]
```diff
@@ -17,7 +17,7 @@ debug_mode
 # Must be root
 root_check
 
-choice=$(whiptail --title "$TITLE" --menu "Which Documentserver do you want to install?\n\nAutomatically configure and install the selected Documentserver.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "Which Documentserver do you want to install?\n\nAutomatically configure and install the selected Documentserver.\n$MENU_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/documentserver.sh:20]: You don't use both here as suggested.

File: menu/fail2ban_menu.sh [hunk 19]
```diff
@@ -25,7 +25,7 @@ else
     STARTUP_SWITCH="OFF"
 fi
 
-choice=$(whiptail --title "$TITLE" --checklist "Automatically install and configure Fail2ban.\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Automatically install and configure Fail2ban.\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/fail2ban_menu.sh:28]: You don't use both here as suggested.

File: menu/documentserver.sh (hunk 18 above)
Comment (enoch85) [inline menu/documentserver.sh:20]: This is good! :+1:

Comment (enoch85) [reply menu/documentserver.sh:20]: Please add `\n\n`

File: menu/fail2ban_menu.sh (hunk 19 above)
Comment (enoch85) [inline menu/fail2ban_menu.sh:28]: This is good, add `\n\n`

File: menu/main_menu.sh [hunk 20]
```diff
@@ -24,7 +24,7 @@ root_check
 ##################################################################
 
 # Main menu
-choice=$(whiptail --title "$TITLE" --menu "Choose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "Choose what you want to do.\n$MENU_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/main_menu.sh:27]: This is good, add `\n\n` :+1:

File: menu/nextcloud_configuration.sh [hunk 21]
```diff
@@ -29,7 +29,7 @@ else
 fi
 
 # Configure Nextcloud
-choice=$(whiptail --title "$TITLE" --checklist "Which settings do you want to configure?\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Which settings do you want to configure?\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/nextcloud_configuration.sh:32]: This is good, add `\n\n` :+1:

File: menu/server_configuration.sh [hunk 22]
```diff
@@ -34,7 +34,7 @@ else
 fi
 
 # Server configurations
-choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to configure\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to configure\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/server_configuration.sh:37]: This is good, add `\n\n` :+1:

File: menu/startup_configuration.sh [hunk 23]
```diff
@@ -43,7 +43,7 @@ The script is smart, and have already pre-selected the values that you'd want to
 fi
 
 # Startup configurations
-choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to change.\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to change.\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/startup_configuration.sh:46]: This is good, add `\n\n` :+1:

File: menu/talk_menu.sh [hunk 24]
```diff
@@ -25,7 +25,7 @@ else
     STARTUP_SWITCH="OFF"
 fi
 
-choice=$(whiptail --title "$TITLE" --checklist "Automatically install and configure Talk.\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Automatically install and configure Talk.\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline menu/talk_menu.sh:28]: This is good, add `\n\n` :+1:

File: network/ddclient-configuration.sh [hunk 25]
```diff
@@ -17,7 +17,7 @@ debug_mode
 # Check if root
 root_check
 
-choice=$(whiptail --title "$TITLE" --menu "This script lets you setup DynDNS by using the ddclient application.\nYou have to setup an account before you can start.\n\nPlease choose your DynDNS-Provider.\n\nIf your DDNS provider isn't already supported, please open a new issue here:\n$ISSUES" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "This script lets you setup DynDNS by using the ddclient application.\nYou have to setup an account before you can start.\n\nPlease choose your DynDNS-Provider.\n$MENU_GUIDE\n\nIf your DDNS provider isn't already supported, please open a new issue here:\n$ISSUES" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline network/ddclient-configuration.sh:20]: You don't add both as suggested.

File: static/main_menu.sh [hunk 26]
```diff
@@ -18,7 +18,7 @@ debug_mode
 root_check
 
 # Main menu
-choice=$(whiptail --title "$TITLE" --menu "Choose what you want to do.\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --menu "Choose what you want to do.\n$MENU_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline static/main_menu.sh:21]: This is good, add `\n\n` :+1:

**Comment by szaimen** [comment]
@enoch85 okay, thanks for the review. I get now what you mean.
So what about changing this text:
`RUN_LATER_GUIDE="You can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"`
to
`RUN_LATER_GUIDE="You can view this script later by running 'sudo bash $SCRIPTS/menu.sh'"`
I guess I woud be fine with adding this to any whiptail-checklist or whiptail-menu then. (at least to all that are not inside the not-supported folder.)

File: static/nextcloud_configuration.sh [hunk 27]
```diff
@@ -21,7 +21,7 @@ debug_mode
 root_check
 
 # Configure Nextcloud
-choice=$(whiptail --title "$TITLE" --checklist "Which settings do you want to configure?\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Which settings do you want to configure?\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline static/nextcloud_configuration.sh:24]: This is good, add `\n\n` :+1:

File: static/server_configuration.sh [hunk 28]
```diff
@@ -33,7 +33,7 @@ debug_mode
 root_check
 
 # Server configurations
-choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to configure\n$CHECKLIST_GUIDE\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE" --checklist "Choose what you want to configure\n$CHECKLIST_GUIDE\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline static/server_configuration.sh:36]: This is good, add `\n\n` :+1:

File: nextcloud_install_production.sh [hunk 29]
```diff
@@ -177,7 +177,7 @@ You could still choose to only run on one disk though, which is not recommended,
 
 You will now get the option to decide which disk you want to use for DATA, or run the automatic script that will choose the available disk automatically."
 
-choice=$(whiptail --title "$TITLE - Choose disk format" --nocancel --menu "How would you like to configure your disks?" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+choice=$(whiptail --title "$TITLE - Choose disk format" --nocancel --menu "How would you like to configure your disks?\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline nextcloud_install_production.sh:180]: You don't add both as suggested.

File: nextcloud_install_production.sh [hunk 30]
```diff
@@ -206,7 +206,7 @@ esac
 # https://unix.stackexchange.com/questions/442598/how-to-configure-systemd-resolved-and-systemd-networkd-to-use-local-dns-server-f    
 while :
 do
-    choice=$(whiptail --title "$TITLE - Set DNS Resolver" --menu "Which DNS provider should this Nextcloud box use?" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    choice=$(whiptail --title "$TITLE - Set DNS Resolver" --menu "Which DNS provider should this Nextcloud box use?\n$MENU_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
```
Comment (enoch85) [inline nextcloud_install_production.sh:209]: You don't add both as suggested.

**Comment by enoch85** [comment]
> Have you looked at the codechanges?

Yes, that's why I wrote as I did.

I made it more obvious now. ;)

**Comment by szaimen** [comment]
@enoch85 have you seen this comment?

> @enoch85 okay, thanks for the review. I get now what you mean.
> So what about changing this text:
> `RUN_LATER_GUIDE="You can view this menu later by running 'sudo bash $SCRIPTS/menu.sh'"`
> to
> `RUN_LATER_GUIDE="You can view this script later by running 'sudo bash $SCRIPTS/menu.sh'"`
> I guess I woud be fine with adding this to any whiptail-checklist or whiptail-menu then. (at least to all that are not inside the not-supported folder.)

**Comment by enoch85** [comment]
> have you seen this comment?

Sorry, I missed that one. But yeah, it looks better. Then it can be used everywhere, and we work against using the `menu.sh` script - always.

**Comment by szaimen** [comment]
@enoch85 I tried to fix all occurences and be very throughly while doing this.

Also I tried to fix the formatting issue (that we have too much text per line in all these whiptails)

**Comment by enoch85** [comment]
Screenshot from 2020-09-25 23:00:00

**Comment by enoch85** [comment]
Very nice structure! Looks much better now.

Perfect! :+1:

**Pull request merged**